modules/compiler/spirv-ll/source/builder_core.cpp

```diff
@@ -919,12 +919,13 @@ cargo::optional<Error> Builder::create<OpConstantNull>(
       if (llvm::cast<llvm::TargetExtType>(type)->getName() == "spirv.Event") {
         constant = llvm::ConstantTargetNone::get(
             llvm::cast<llvm::TargetExtType>(type));
+        break;
       }
-      break;
+      [[fallthrough]];
```

Comment:
It changes the behaviour in how that it will hit an 'unreachable', whereas before it would deference a null pointer.

This is okay since `OpConstantNull` is only valid on a limited range of types (see [the spec](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#OpConstantNull)) and out of these, we'd only expect events to be target extension types. Anything else is unhandled, as we do with device events, queues, etc.

Reply:
I can chuck this into the commit description as I may have left it a bit unclear, sorry

Reply:
sure, I just thought it worth checking.